tests/testthat/test-events.R

```diff
@@ -0,0 +1,293 @@
+    "human", states = list(states$E, states$IMild)))
+
+  parameters <- list(human_population = 100000, average_age = 27)
+
+  initial_age <- trunc(
+    r_exp(parameters$human_population, parameters$average_age))
+
+  variables <- list(
+    age <- individual::Variable$new("age", 50 - initial_age))
+
+
+  api <- mock_api(list(), parameters = parameters)
+
+  create_event_based_processes(individuals, states, variables, events,
+                               parameters)
+
+  # test a few listeners: test they don't call for an empty target
+  events$exposure$listeners[[1]](api, numeric(0))
+  mockery::expect_called(api$queue_state_update, 1)
+
+  events$mild_infection$listeners[[2]](api, numeric(0))
+  mockery::expect_called(api$queue_state_update, 1)
+
+})
+
+test_that("test create_infection_update_listener", {
+
+  human <- mockery::mock()
+  IMild <- mockery::mock()
+
+  ret <- create_infection_update_listener(
+    human,
+    IMild
+  )
+
+  api <- list(queue_state_update = mockery::mock())
+  to_move <- mockery::mock()
+  ret(api, to_move)
+  mockery::expect_args(api$queue_state_update, 1, human, IMild, to_move)
+
+})
+
+test_that("test create_progression_listener", {
+
+  events <- list(exposure = mockery::mock())
+  shift <- 0
+  duration <- 1
+  target <- mockery::mock()
+  func_mock <- mockery::mock()
+  r_erlang_mock <- mockery::mock(c(TRUE, TRUE, TRUE, TRUE))
+
+  ret <- create_progression_listener(
+    events$exposure,
+    duration,
+    shift,
+    func_mock
+  )
+
+  api <- list(schedule = mockery::mock())
+
+  mockery::stub(ret, 'r_erlang', mockery::mock(c(TRUE, TRUE, TRUE, TRUE)))
+
+  with_mock(
+    'hypatia::r_erlang' = r_erlang_mock,
+    ret2 <- api$schedule(
+      events$exposure,
+      target,
+      rep(.5, 4)
+    )
+  )
+
+  ret(api, target)
+
+  mockery::expect_args(
+    api$schedule,
+    1,
+    event = events$exposure,
+    target = target,
+    func_mock = c(0.5, 0.5, 0.5, 0.5)
+  )
+
+  mockery::expect_called(api$schedule, 2)
+
+})
+
+test_that("test create_exposure_update_listener for hosp TRUE", {
+
+  human <- mockery::mock()
+  states <- mockery::mock()
+
+  ret <- create_exposure_update_listener(
+    human,
+    states,
+    events,
+    variables,
+    parameters
+  )
+
+  api <- list(get_variable = mockery::mock(), schedule = mockery::mock())
+  variables <- list(discrete_age = mockery::mock())
+  events <- list(exposure = mockery::mock())
+  problist <- seq(1,17,1)
+  exposure = mockery::mock()
+  parameters <- list(dur_E = mockery::mock(),
+                     prob_hosp = list(mockery::mock(problist)))
+  to_move <- c(6, 7, 3, 2)
+  discrete_age <- mockery::mock()
+  target <- 100
+
+  r_erlang_mock <- mockery::mock(c(TRUE, TRUE, TRUE, TRUE))
+  mockery::stub(ret, 'r_erlang', mockery::mock(c(TRUE, TRUE, TRUE, TRUE)))
+
+  mockery::stub(ret, 'bernoulli_multi_p', mockery::mock(TRUE))
+
+  mockery::stub(
+    ret,
+    "hosp",
+    mockery::mock()(
```

Comment:
I'm not sure what we're trying to do here but it doesn't look like the correct use of `mockery::mock`. I'll ask offline